File: src/main/java/de/codehat/signcolors/commands/CommandManager.java
/*
 * Copyright (c) 2017 CodeHat.
 * This file is part of 'SignColors' and is licensed under GPLv3.
 */

package de.codehat.signcolors.commands;

import de.codehat.signcolors.SignColors;
import de.codehat.signcolors.util.Message;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

import java.util.HashMap;

public class CommandManager implements CommandExecutor {

    private SignColors plugin;
    private HashMap<String, AbstractCommand> commandDatabase = new HashMap<>();

    public CommandManager(SignColors plugin) {
        this.plugin = plugin;
    }

    public void registerCommand(String command, AbstractCommand object) {
        commandDatabase.put(command, object);
    }

    private boolean exists(String name) {
        return commandDatabase.containsKey(name);
    }

    private AbstractCommand getExecutor(String name) {
        return commandDatabase.get(name);
    }

    @Override
    public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
        if (args.length == 0) {
            // Execute info command
            this.getExecutor("").onCommand(sender, cmd, label, args);
        } else if (exists(args[0])) {
            // Execute one of the other commands
            this.getExecutor(args[0]).onCommand(sender, cmd, label, args);
        } else {
            Message.sendWithLogo(sender, this.plugin.getStr("UNKNOWNCMD"));
            Message.sendWithLogo(sender, this.plugin.getStr("UNKNOWNCMDH"));
        }
        return true;
    }
}

File: src/main/kotlin/de/codehat/signcolors/command/CommandManager.kt
/*
 * SignColors is a plug-in for Spigot adding colors and formatting to signs.
 * Copyright (C) 2022 CodeHat
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package de.codehat.signcolors.command

import de.codehat.signcolors.language.LanguageKey
import de.codehat.signcolors.util.sendLogoMsg
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

class CommandManager : CommandExecutor {

    private val commands: MutableMap<String, Command> = mutableMapOf()

    companion object {
        const val CMD_PREFIX = "sc"
        const val CMD_INFO = ""
        const val CMD_HELP = "help"
        const val CMD_RELOAD = "reload"
        const val CMD_COLOR_CODES = "colorcodes"
        const val CMD_GIVE_SIGN = "givesign"
        const val CMD_MIGRATE_DATABASE = "migratedb"
    }

    override fun onCommand(
        sender: CommandSender,
        command: org.bukkit.command.Command,
        label: String,
        args: Array<out String>
    ): Boolean {
        when {
            args.isEmpty() -> commands[""]?.onCommand(sender, command, label, args)
            commands.containsKey(args[0]) ->
                commands[args[0]]?.onCommand(sender, command, label, args)
            else -> sender.sendLogoMsg(LanguageKey.UNKNOWN_CMD)
        }
        return true
    }

    fun registerCommand(command: String, commandObject: Command): Boolean {
        return commands.put(command, commandObject) == null
    }
}
